0000-settlement-engines/0000-settlement-engines.md

```diff
@@ -0,0 +1,395 @@
+
+Operators and external factors outside the control of the settlement engine implementation may also cause inconsistences, such as network connectivity, peers operating incompatible settlement engines, or malicious peers.
+
+## Exchanging messages
+
+In order to settle or receive settlements with a peer, a settlement engine may first need to retrieve or communicate information with the peer's settlement engine. Two peered settlement engine instances may send and receive settlement-related messages among themselves, such as identifiers for their ledger accounts.
+
+To support multiple interoperable settlement engine implementations for a particular settlement system, implementators may standardize the schema and type of messages their settlement engines use to communicate with one another. This work is out-of-scope of this RFC.
+
+Interledger connectors use a transport, such as HTTP or WebSockets, to send and receive data with peers. Settlement engine implementations SHOULD proxy all messages through its Interledger connector's existing transport like so:
+
+1. Origin settlement engine sends a request to its connector with the settlement-related message to forward.
+2. Origin connector encodes the raw message within an ILP Prepare packet (described below), which is sent to the peer's connector using its existing transport.
+3. Peer connector receives the message within the ILP Prepare, identifies which settlement engine instance the account is associated with, and sends a request to its settlement engine to handle the message. The peer connector MUST NOT forward the ILP Prepare to any other connectors.
+4. Peer settlement engine processes the message and responds with its own message.
+5. Peer connector sends the response message back across the transport to the origin connector within an ILP Fulfill or ILP Reject, depending upon the code of the response (described below). If the peer, connector was unable to process the request, it MUST respond with an ILP Reject.
+6. Origin connector sends the response message back to the origin settlement engine.
+
+### ILP Prepare
+
+- `amount`: `0`
+- `expiresAt`: _Determined by connector_
+- `executionCondition`: `e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`
+- `destination`: `peer.settle`
+- `data`: _Request message from sender settlement engine_
+
+### ILP Fulfill
+
+- `fulfillment`: `0000000000000000000000000000000000000000000000000000000000000000`
+- `data`: _Response message from recipient settlement engine_
+
+### ILP Reject
+
+- `code`: _Determined by connector from HTTP status of forwarded request_
+- `triggeredBy`: `peer.settle`
+- `message`: _Determined by connector_
+- `data`: _Response message from recipient settlement engine, or empty if antecedent failure_
+
+## Accounts and identifiers
+
+Each account MUST be identified by a unique, [URL-safe](https://tools.ietf.org/html/rfc3986#section-2.3) string, immutable for the lifetime of the account.
```

Comment:
Is this referring to settlement engine accounts?
```suggestion
Each settlement engine account MUST be identified by a unique, [URL-safe](https://tools.ietf.org/html/rfc3986#section-2.3) string. Account identifiers MUST be immutable for the lifetime of the account.
```

Reply:
> Account identifiers MUST be immutable for the lifetime of the account.

Is this true, or should this be a `SHOULD`?

Reply:
For safety, I think we should leave this as a `MUST`; changing identifiers introducing a lot of complexity.